### PR TITLE
front: Add Dropdown

### PR DESCRIPTION
**Pull request by helaan**
This is a cherry-pick of 2385175, 26f054f and a partial cherry-pick of 21069da and 541f4ca

### CODE REVIEW

File: front/src/Dropdown.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import Dropdown from "./Dropdown";
+import renderer from "react-test-renderer";
+
+test("Dropdowns can be changed", () => {
```
Comment (bafonins) [inline front/src/Dropdown.test.js]: use `describe` and `it` blocks.

File: front/src/Dropdown.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ *  - onChange: a function to call when a selection is made
+ *  - enabled: if the dropdown should be enabled
+ */
+const Dropdown = props => {
+  const { currentValue, values, onChange, enabled } = props;
+  return (
+    <select
+      value={currentValue}
+      onChange={event => {
+        if (enabled) onChange(event.target.value);
```
Comment (bafonins) [inline front/src/Dropdown.js]: why checking for `enabled` ?

Comment (helaan) [reply front/src/Dropdown.js]: otherwise the second test fails

File: front/src/Dropdown.test.js (hunk 1 above)
Comment (helaan) [reply front/src/Dropdown.test.js]: fixed

File: front/src/Dropdown/index.test.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    );
+    let tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+
+    // Check if we can't trigger onChange
+    expect(value).toBe("tea");
+    tree.props.onChange({ target: { value: "coffee" } });
+    expect(value).toBe("tea");
+  });
+});
```
Comment (bafonins) [inline front/src/Dropdown/index.test.js]: Just matching a snapshot should be enough for testing this component. I dont think you have to test the `disabled` html attribute.

**Pull request merged**